src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Arcus.Messaging.Pumps.Abstractions
+{
+    public abstract class MessagePump<TMessage, TMessageContext> : BackgroundService
+        where TMessageContext : MessageContext
+    {
+        /// <summary>
+        ///     Logger to write telemetry to
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        ///     Configuration of the application
+        /// </summary>
+        protected IConfiguration Configuration { get; }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        protected MessagePump(IConfiguration configuration, ILogger logger)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNull(configuration, nameof(configuration));
+
+            Logger = logger;
+            Configuration = configuration;
+        }
+
+        /// <summary>
+        ///     Handles an exception that occured during the receiving of a message
+        /// </summary>
+        /// <param name="receiveException">Exception that occured</param>
+        protected Task HandleReceiveExceptionAsync(Exception receiveException)
+        {
+            Logger.LogError(receiveException, "Unable to process message");
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        ///     Deserializes a raw JSON message body
+        /// </summary>
+        /// <param name="rawMessageBody">Raw message body to deserialize</param>
+        /// <param name="messageContext">Context concerning the message</param>
+        /// <returns>Deserialized message</returns>
+        protected TMessage DeserializeJsonMessageBody(byte[] rawMessageBody, MessageContext messageContext)
+        {
+            var encoding = DetermineMessageEncoding(messageContext);
+            var serializedMessageBody = encoding.GetString(rawMessageBody);
```

Comment:
don't use var

Reply:
It's clear that it's a string, so no need according to Arcus guidelines.